app/src/main/java/com/example/tasky/feature_authentication/presentation/login_screen/LoginScreen.kt

```diff
@@ -0,0 +1,182 @@
+                }
+                is AuthUseCaseResult.Success -> {
+                    TODO("Navigate to Agenda Screen")
+                }
+            }
+        }
+    }
+
+    Column(
+        horizontalAlignment = Alignment.CenterHorizontally,
+        modifier = Modifier
+            .fillMaxSize()
+            .background(BackgroundBlack)
+    ) {
+        Text(
+            text = stringResource(id = R.string.WelcomeMessage),
+            color = Color.White,
+            fontSize = 28.sp,
+            modifier = Modifier.padding(top = 77.dp)
+        )
+        Spacer(modifier = Modifier.height(72.dp))
+        Box(modifier = Modifier
+            .fillMaxSize()
+            .clip(
+                RoundedCornerShape(
+                    topStart = 30.dp,
+                    topEnd = 30.dp
+                )
+            )
+            .background(BackgroundWhite)
+        ) {
+            Column(
+                horizontalAlignment = Alignment.CenterHorizontally,
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(
+                        top = 50.dp,
+                        start = 17.dp,
+                        end = 15.dp
+                    )
+            ) {
+                TextFieldComponent(
+                    value = state.email,
+                    onValueChanged = {
+                        viewModel.onEvent(LoginEvent.EmailChanged(it))
+                    },
+                    hint = stringResource(id = R.string.EmailHint),
+                    keyboardType = KeyboardType.Email,
+                    isValid = state.isEmailValid,
+                    contentDescription = stringResource(id = R.string.DescriptionEmailValid)
+                )
+                Spacer(modifier = Modifier.height(15.dp))
+                TextFieldComponent(
+                    value = state.password,
+                    onValueChanged = {
+                        viewModel.onEvent(LoginEvent.PasswordChanged(it))
+                    },
+                    hint = stringResource(id = R.string.PasswordHint),
+                    keyboardType = KeyboardType.Password,
+                    passwordVisible = state.isPasswordVisible,
+                    onClick = {
+                        viewModel.onEvent(LoginEvent.ChangePasswordVisibility)
+                    },
+                    contentDescription = stringResource(id = R.string.DescriptionPasswordVisibility)
+                )
+                Spacer(modifier = Modifier.height(25.dp))
+                Button(
+                    onClick = {
+                        viewModel.onEvent(LoginEvent.SignIn)
+                    },
+                    colors = ButtonDefaults.buttonColors(
+                        containerColor = BackgroundBlack
+                    ),
+                    modifier = Modifier
+                        .width(324.dp)
```

Comment:
This doesn't scale up on larger devices which would look weird there. `fillMaxWidth` + padding would make more sense here